### PR TITLE
fix(attachment): themeing and redline

### PR DESCRIPTION
**Pull request by bcalvery**
This is an update to add dark and contrast theme files for Attachment and update the styling to match redlines.

Current problems with pxToRem mean that the these changes will not render correctly until pxToRem is fixed.

Before:
image

After:
**Default (rest / hover / focus states)**
image

**Dark (rest / hover / focus states)**
image

**High contrast (rest / hover /focus states)**
image

### CODE REVIEW

File: docs/src/examples/components/Attachment/Variations/AttachmentActionableExample.shorthand.tsx [hunk 1]
```diff
@@ -2,18 +2,21 @@ import * as React from 'react'
...
         actionable
-        icon="file word outline"
+        icon="table"
         header="Document.docx"
         description="800 Kb"
-        action={{ icon: 'x' }}
+        action={{ icon: 'more', onClick: this.handleClick('More Action') }}
         progress={33}
-        onClick={this.handleClick}
+        onClick={this.handleClick('Attachment')}
```
Comment (layershifter) [inline docs/src/examples/components/Attachment/Variations/AttachmentActionableExample.shorthand.tsx:25]: We should not use `alert()` in docs, can we implement logging like it is done there?

https://stardust-ui.github.io/react/components/portal#types-portal
https://github.com/stardust-ui/react/edit/master/docs/src/examples/components/Portal/Types/PortalExample.shorthand.tsx?message=docs(PortalExample.shorthand):%20your%20description

Comment (bmdalex) [reply docs/src/examples/components/Attachment/Variations/AttachmentActionableExample.shorthand.tsx:25]: @layershifter we do have examples that use `alert()`; maybe we should add a chore task to replace all with logging

File: packages/react/src/themes/teams-high-contrast/components/Attachment/attachmentVariables.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import { AttachmentVariables } from '../../../teams/components/Attachment/attachmentVariables'
+
+export default (siteVariables: any): Partial<AttachmentVariables> => {
+  return {
+    border: `1px solid ${siteVariables.white}`,
+    backgroundColor: siteVariables.black,
+    backgroundColorHover: siteVariables.accessibleYellow,
+    textColor: siteVariables.white,
+    textColorHover: siteVariables.black,
+    boxShadow: '',
```
Comment (bmdalex) [inline packages/react/src/themes/teams-high-contrast/components/Attachment/attachmentVariables.ts]: `''` (empty string) is not an accepted CSS value; can we use undefined instead?

File: packages/react/src/themes/teams/components/Attachment/attachmentVariables.ts [hunk 3]
```diff
@@ -22,24 +28,31 @@ export type AttachmentVariables = {
...
+    iconSpace: pxToRem(12),
+    border: `1px solid ${siteVariables.gray08}`,
+    borderRadius: '3px',
+    backgroundColor: siteVariables.gray10,
+    backgroundColorHover: siteVariables.gray08,
+    textColor: siteVariables.colors.grey[900],
+    textColorHover: siteVariables.colors.grey[900],
+    boxShadow: siteVariables.shadowLevel1,
+
+    actionColorContrastOverride: '',
```
Comment (bmdalex) [inline packages/react/src/themes/teams/components/Attachment/attachmentVariables.ts]: pls use undefined instead of empty strings

**Review by bmdalex — APPROVED** [review]

File: packages/react/src/themes/teams-dark/components/Attachment/attachmentVariables.ts [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import { AttachmentVariables } from '../../../teams/components/Attachment/attachmentVariables'
+
+export default (siteVariables: any): Partial<AttachmentVariables> => {
+  return {
+    textColor: siteVariables.colors.grey[50],
+    textColorHover: siteVariables.colors.grey[50],
+  }
+}
```
Comment (layershifter) [inline packages/react/src/themes/teams-dark/components/Attachment/attachmentVariables.ts]: nit, can be simplified to:

```
export default (siteVariables: any): Partial<AttachmentVariables> => ({
  textColor: siteVariables.colors.grey[50],
  textColorHover: siteVariables.colors.grey[50],
})
```

File: packages/react/src/themes/teams-high-contrast/components/Attachment/attachmentVariables.ts [hunk 5]
```diff
@@ -0,0 +1,19 @@
+import { AttachmentVariables } from '../../../teams/components/Attachment/attachmentVariables'
+
+export default (siteVariables: any): Partial<AttachmentVariables> => {
+  return {
+    border: `1px solid ${siteVariables.white}`,
```
Comment (layershifter) [inline packages/react/src/themes/teams-high-contrast/components/Attachment/attachmentVariables.ts]: pxToRem?

File: packages/react/src/themes/teams/components/Attachment/attachmentStyles.ts [hunk 6]
```diff
@@ -15,9 +18,16 @@ const attachmentStyles: ComponentSlotStylesInput<AttachmentProps, AttachmentVari
...
     color: v.textColor,
+    boxShadow: v.boxShadow,
+    border: v.border,
+    borderRadius: v.borderRadius,
 
     outline: 0,
 
+    '& *': {
+      boxSizing: `border-box`,
+    },
```
Comment (layershifter) [inline packages/react/src/themes/teams/components/Attachment/attachmentStyles.ts:31]: Please add a TODO for #1057.

File: packages/react/src/themes/teams/components/Attachment/attachmentVariables.ts [hunk 7]
```diff
@@ -22,24 +28,31 @@ export type AttachmentVariables = {
...
-  descriptionLineHeight: siteVariables.lineHeightSmall,
-
-  focusOutlineColor: siteVariables.colors.primary[500],
-})
+export default (siteVariables: any): AttachmentVariables => {
+  return {
+    padding: `${pxToRem(7)} ${pxToRem(3)} ${pxToRem(7)} ${pxToRem(11)}`, // padding set to 1px less to account for 1px border
+    iconSpace: pxToRem(12),
+    border: `1px solid ${siteVariables.gray08}`,
+    borderRadius: '3px',
```
Comment (layershifter) [inline packages/react/src/themes/teams/components/Attachment/attachmentVariables.ts]: pxToRem?

**Review by codepretty — APPROVED** [review]

**Review by layershifter — APPROVED** [review]

**Pull request merged**